straxen/corrections_services.py

```diff
@@ -0,0 +1,222 @@
+"""Return corrections from corrections DB
+"""
+from datetime import datetime
+from datetime import timezone
+import pytz
+from bson.son import SON
+import pymongo
+import numpy as np
+from warnings import warn
+
+import strax
+import straxen
+from .rundb import default_mongo_url, backup_mongo_urls
+
+export, __all__ = strax.exporter()
+
+
+@export
+class CmtServices():
+    """
+    A class that returns corrections
+    Corrections are set of parameters to be applied in the analysis
+    stage to remove detector effects. Information on the strax implementation
+    can be found at https://github.com/AxFoundation/strax/blob/master/strax/corrections.py
+    """
+    def __init__(self, host=None, username='nt_analysis', password=None):
+        """
+        :param host: DB host
+        :param username: DB username
+        :param password: DB password
+        """
+        self.host = host
+        self.username = username
+        self.password = password
+        if self.username.endswith('analysis'):
+            self.password = straxen.get_secret('rundb_password')
+
+        if host is None:
+            mongo_connections = [default_mongo_url, *backup_mongo_urls]
+            for url in mongo_connections:
+                try:
+                    mongo_url = f'mongodb://{url}'
+
+                    self.interface = strax.CorrectionsInterface(
+                        host=mongo_url,
+                        username=self.username,
+                        password=self.password,
+                        database_name='corrections')
+                    break
+                except pymongo.errors.ServerSelectionTimeoutError:
+                    warn(f'Cannot connect to to Mongo url: {url}')
+                    if url == mongo_connections[-1]:
+                        raise pymongo.errors.ServerSelectionTimeoutError(
+                            'Cannot connect to any Mongo url')
+        else:
+            self.interface = strax.CorrectionsInterface(
+                    host=self.host,
+                    username=self.username,
+                    password=self.password,
+                    database_name='corrections')
+
+    def get_corrections_config(self, run_id, correction=None, config_model=None):
+        """
+        Get context configuration for a given correction
+        :param run_id: run id from runDB
+        :param correction: correction's name (str type)
+        :param config_model: configuration model (dict type)
+        :return: correction value(s)
+        """
+
+        if not isinstance(config_model, tuple):
+            raise ValueError(f'config_model {config_model} must be a tuple')
+        if len(config_model) == 3:
```

Comment:
This looks like it must be a tuple, but I think you are right in the doc-string a dict would be a more elegant solution here. 

You could test if the keys are specified and if not add the corresponding thing as a default e.g.

```
if not xeno1t in config_model.keys():
    config_model[xenon1t] = Flase
```

Reply:
could be a more elegant solution, I will revise this later as Chris says "premature optimization root of all evil"